docs/DeveloperGuide.md

```diff
@@ -96,10 +96,37 @@ Below is the Sequence Diagram for interactions within the `Logic` component for
 The `Model`:
 
 * stores a `UserPref` object that represents the user’s preferences.
-* stores the finance tracker data.
+* stores the finance tracker data in a `TransactionList` containing `Transaction`s.
 * exposes `MonthlySavings` and `MonthlyBudget`, which can be 'observed' e.g. the UI can be bound to the values in these classes so that the UI automatically updates when the values in the classes change.
 * exposes an unmodifiable `ObservableList<Transaction>` which can be 'observed' e.g. the UI can be bound to this list so that the UI automatically updates when the data in the list changes.
-* does not depend on any of the other three components.
+* does not depend on any of the other three components (`UI`, `Logic`, `Storage`).
+
+A `Transaction`:
+* represents a unit of user data within the finance tracker.
+* is either an `Expense` or an `Income`.
+* is composed of a `Title`, an `Amount`, a `Date`, and any number of `Category`s. These are known as *data fields*.
+
+The *data fields* `Title`, `Amount`, `Date`, and `Category` are encapsulations of an underlying Java data type.
+
+| Data Field  | Underlying Java Data Type   |
+| ----------- | -------------------------   |
+| `Title`     | `java.lang.String`          |
+| `Amount`    | `java.math.BigDecimal`      |
+| `Date`      | `java.time.LocalDate`       |
+| `Category`  | `java.lang.String`          |
+
+The underlying Java data types allow more operations to be done on `Transaction` objects, such as filtering transactions by date, or aggregating the amounts of expenses and incomes.
```

Comment:
Unsure if the keywords e.g. "transaction", "date", "amount", "expense", "income" should reference the class names.